File: ruby/tutorials/1-Getting-Started/my_first_script.rb
#!/usr/bin/env ruby
# My First Ruby Script

# Print a greeting
puts "Hello! This is my first Ruby script."

# Variables
my_name = "Ruby Learner"
favorite_color = "blue"

puts "My name is #{my_name}"
puts "My favorite color is #{favorite_color}"

# A simple calculation
age = 25
puts "In 5 years, I will be #{age + 5} years old"

# Working with arrays
hobbies = ["reading", "coding", "gaming"]
puts "\nMy hobbies:"
hobbies.each do |hobby|
  puts "  - #{hobby}"
end

# A simple method
def favorite_quote
  "The only way to do great work is to love what you do."
end

puts "\nFavorite quote: #{favorite_quote}"

File: dart/tutorials/1-Getting-Started/my_first_script.dart
// My first Dart script
// This simple script demonstrates basic Dart syntax

void main() {
  // Print a welcome message
  print('Welcome to Dart programming!');

  // Variables
  var name = 'Dart Learner';
  var age = 25;

  // String interpolation
  print('Hello, $name!');
  print('You are $age years old.');

  // Basic arithmetic
  var nextYear = age + 1;
  print('Next year you will be $nextYear years old.');

  // Lists
  var languages = ['Dart', 'Python', 'JavaScript', 'Ruby'];
  print('\nYou are learning: $languages');
  print('First language: ${languages.first}');

  // Functions
  print('\nCalculating...');
  var sum = add(5, 3);
  print('5 + 3 = $sum');

  print('\n🎉 Congratulations on running your first Dart script!');
}

// A simple function
int add(int a, int b) {
  return a + b;
}
